### ISSUE TITLE
unable to verify session for vc

### ISSUE THREAD

**Issue by tormig-softronic** [vmware-tanzu/vsphere-kubernetes-drivers-operator#151]
Installed the operator from OperatorHub in an Openshift 4.9 cluster. The vspherecloudconfig instance failes with these messages:

- in the yaml:
  status:
    config: failed
    message: unable to verify session for vc \<vcIp\>
- in the controller pod log:
  E0407 10:43:24.522831       1 controller.go:302] controller-runtime/manager/controller/vdoconfig "msg"="Reconciler error"     "error"="Failed to verify vSphere connection information. Status of VsphereCloudConfig resource is in failed state" "name"="compat-matrix-config" "namespace"="vsphere-kubernetes-drivers-operator" "reconciler group"="vdo.vmware.com" "reconciler kind"="VDOConfig"

This also appears depending on which options you choose:
"E0408 07:15:38.968740       1 vspherecloudconfig_controller.go:66] controllers/vSphereCloudConfig "msg"="error occurred when reconciling vSphere credentials" "error"="unable to verify session for vc \<vcIp\>: ServerFaultCode: Permission to perform this operation was denied." "name"="vspherecloudconfig" "namespace"="vsphere-kubernetes-drivers-operator" "vcIp"="\<vcIp\>"

This vSphere user account has Read-only permission on the vCenter level  and is Administrator on the Datacenter involved (and it's children). The same user works for the In-tree vSphere provider in the same cluster.

**Comment by lgayatri** [comment]
Is this issue no longer seen ?

**Comment by tormig-softronic** [comment]
Sorry, needed some more investigations. We have have followed this guide: https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-0AB6E692-AA47-4B6A-8CEA-38B754E16567.html but still no success. 

As mentioned before we started with the user having the Administrator role at the Datacenter level (This object and its children) and now we replaced the Read-only role with the "CNS-SEARCH-AND-SPBM" role in the guide for this user at vCenter Server level (This object). It still fails

We did a brief test with the Administrator role at the vCenter Server level and then it worked of course but we cannot allow this user to have this role at the top level.

**Comment by bortek** [comment]
I am also seeing this at the first step when setting up VsphereCloudConfig in Openshift.

**Comment by jvrahav** [comment]
can you ensure you have enabled `Sessions` privilege on the role that you are using here.

**Comment by bortek** [comment]
I went from using operator to use the latest driver setup manually and now implementing gitops style approach to sync config via Argocd. The operator in any way does not have the latest and greatest driver version.

**Comment by tormig-softronic** [comment]
@jvrahav we applied it to the CSN-SEARCH-AND-SPBM role in guide and then it works. Something VMware support not yet has figured out. Thank you. 

(and now we a new problem as a result, vsphere-csi-node, but i suppose that will be a new issue)
